src/git/repo.rs

```diff
@@ -1446,6 +1672,98 @@ impl<'repo> Branch<'repo> {
     }
 }
 
+/// A Git file status indicator.
+/// See https://git-scm.com/docs/git-status#_short_format
+#[allow(missing_docs)]
+#[derive(Debug, PartialEq, Eq)]
+pub enum FileStatus {
+    Unmodified,
+    Modified,
+    Added,
+    Deleted,
+    Renamed,
+    Copied,
+    Unmerged,
+    Untracked,
+    Ignored,
+}
+
+/// The status of a file in the repo.
+#[derive(Debug, PartialEq, Eq)]
+pub struct StatusEntry {
+    /// The status of the file in the index.
+    pub index_status: FileStatus,
+    /// The status of the file in the working tree.
+    pub work_tree_status: FileStatus,
+    /// The file path.
+    pub path: String,
+    /// The original path of the file (for renamed files).
+    pub orig_path: Option<String>,
+}
+
+impl StatusEntry {
+    /// Returns the paths associated with the status entry.
+    pub fn paths(&self) -> Vec<PathBuf> {
+        let mut result = vec![PathBuf::from(self.path.clone())];
+        if let Some(orig_path) = &self.orig_path {
+            result.push(PathBuf::from(orig_path.clone()));
+        }
+        result
+    }
+}
+
+fn parse_file_status(status: char) -> eyre::Result<FileStatus> {
+    let status = match status {
+        ' ' => FileStatus::Unmodified,
+        'M' => FileStatus::Modified,
+        'A' => FileStatus::Added,
+        'D' => FileStatus::Deleted,
+        'R' => FileStatus::Renamed,
+        'C' => FileStatus::Copied,
+        'U' => FileStatus::Unmerged,
+        '?' => FileStatus::Untracked,
+        '!' => FileStatus::Ignored,
+        _ => eyre::bail!("invalid status indicator: {}", status),
```

Comment:
Let's just `warn!` in this case and use a sensible default (maybe `FileStatus::Untracked` or `FileStatus::Ignored`?).